File: server/php/src/Application/Domain/Payments/PaymentDetails/SurchargeRate.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class SurchargeRate
{
    public ?float $adValoremRate = null;
    public ?int $specificRate = null;
    public ?string $surchargeProductTypeId = null;
    public ?string $surchargeProductTypeVersion = null;

    public function toArray(): array
    {
        return [
            'adValoremRate' => $this->adValoremRate,
            'specificRate' => $this->specificRate,
            'surchargeProductTypeId' => $this->surchargeProductTypeId,
            'surchargeProductTypeVersion' => $this->surchargeProductTypeVersion,
        ];
    }
}

File: server/java/src/main/java/com/onlinepayments/example/service/domain/paymentdetails/SurchargeRate.java
package com.onlinepayments.example.service.domain.paymentdetails;

import java.math.BigDecimal;

public record SurchargeRate(
    BigDecimal adValoremRate,
    Integer specificRate,
    String surchargeProductTypeId,
    String surchargeProductTypeVersion
) { }
